model-optimizer/extensions/back/MarkNodesWithShapeValues.py

```diff
@@ -52,27 +53,48 @@ def get_operations_with_shape_inputs():
             'OneHot': [1],  # depth input
         }
 
-    def find_and_replace_pattern(self, graph: Graph):
-        shape_input_ops_map = self.get_operations_with_shape_inputs()
-
-        nodes_with_shape_inputs = []
+    @staticmethod
+    def get_nodes_with_shape_inputs(graph: Graph) -> List[Node]:
+        shape_accepting_ops = MarkNodesWithShapeValues.get_shape_accepting_ops()
+        shape_accepting_nodes = []
         for node in graph.get_op_nodes():
-            if node.soft_get('type') in shape_input_ops_map:
-                nodes_with_shape_inputs.append(node)
+            if node.soft_get('type') in shape_accepting_ops:
+                shape_accepting_nodes.append(node)
+        return shape_accepting_nodes
 
-        start_nodes = []
+    @staticmethod
+    def get_sources_for_nodes(nodes_with_shape_inputs: List[Node], condition: Callable) -> List[Node]:
+        shape_accepting_ops = MarkNodesWithShapeValues.get_shape_accepting_ops()
```

Comment:
I would recommend to add description for this method: brief, parameters description. Also, please consider to rename `condition`, not clear what condition will check according to this logic.

Reply:
Agree that `condition` should be removed. About description, I don't know I thought that such short functions look self explanatory, but let's discuss during online review

Reply:
renamed condition, but leaved functions without comments. I think they look self explanatory. @lazarevevgeny what do you think?